File: python/src/ev1/ut1_fundamentospython/u02tiposvariables/teoria/U01_VariablesIntro.py
"""
U01 — Variables y tipos basicos en Python.

En esta primera clase de variables aprenderemos los conceptos básicos
que giran en torno a las mismas, ¿como se escriben? ¿como funcionan?
¿que puedo guardar en ellas? Y así nos abriremos paso poco a poco en
este nuevo lenguaje.

OBJETIVO:
  - Crear variables (nombre = valor).
  - Conocer tipos: int, float, str, bool.
  - Mostrar valores con print y f-strings.

REGLAS:
  - Nombres en minusculas_con_guion_bajo (snake_case).

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
"""

# Diferentes tipos de datos de ejemplo.
numero_entero = 14
numero_decimal = 3.1416
cadena_texto = "Agustín"
soy_humano = True
hoy_lunes = False

# Variables de números enteros (int)
edad_profesor = 26
dientes = 32
amigos = 5

# Variables de números decimales (double)
altura = 1.76
saldo_banco = 1250.35
precio_hot_dog_costco = 1.50

# Variables de texto (str)
dia_semana = "Martes"
dia_actual = "7"
mes_actual = "Julio"
año_actual = "2026"
mensaje = "¿Cuando es tú cumpleaños?"

# Variables de booleanos (bool)
lloviendo = False
gane_loteria = False
tengo_mascotas = True

# Imprimir los datos de ejemplo usando "print()"

print("\nINICIO DEL PROGRAMA")

#prin("               Trozo 1         " +    Trozo 2   + "   Trozo 3    ")
print("\n ¡Soy tú profesor! Mi nombre es " + cadena_texto + ", mucho gusto. \n")

print("\n ¡IMPRIMIENDO VARIABLES DE NÚMEROS ENTEROS! \n")
print(f"Tú profesor tiene {edad_profesor} años.")
print(f"Tú profesor tiene {dientes} dientes.")
print(f"Tú profesor tiene {amigos} amigos.")

print("\n ¡IMPRIMIENDO VARIABLES DE NÚMEROS DECIMALES! \n")
print(f"Tú profesor mide {altura} centimetros.")
print(f"Tú profesor tiene {saldo_banco} euros en el banco.")
print(f"Tú profesor compro un Hot Dog en Costco por {precio_hot_dog_costco}")

print("\n ¡IMPRIMIENDO VARIABLES DE TEXTO! \n")
print("Hoy es " + dia_semana + ", " + dia_actual + " de " + mes_actual + " del " + año_actual + ".")
print(f"Hoy es {dia_semana}, {dia_actual} de {mes_actual} del {año_actual}.")
print("El nombre de tú profesor es " + cadena_texto + ".")

print("\n ¡IMPRIMIENDO VARIABLES DE BOOLEANOS! \n")
print(f"¿Hoy esta lloviendo? La respuesta es: {lloviendo}")
print(f"¿Me he ganado la loteria? La respuesta es: {gane_loteria}")
print(f"¿Tengo mascotas? La respuesta es: {tengo_mascotas}")

print("\nFIN DEL PROGRAMA\n")

File: csharp/src/ev1/ut1_fundamentoscsharp/u02tiposvariables/teoria/U01_VariablesIntro.cs
/*
U01 — Variables y tipos basicos en C#.

En esta primera clase de variables aprenderemos los conceptos basicos:
¿como se declaran? ¿como funcionan? ¿que puedo guardar en ellas?

OBJETIVO:
  - Declarar variables con tipo explicito (int, double, string, bool, char).
  - Asignar valores y mostrarlos con Console.WriteLine.
  - Usar interpolacion $"..." para mezclar texto y variables.
  - Conocer camelCase como convencion de nombres en C#.

REGLAS:
  - Toda variable tiene un tipo fijo (a diferencia de Python).
  - Cada instruccion termina en punto y coma (;).

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

Console.WriteLine("\n¡INICIO DEL PROGRAMA!\n");

// --- Datos de ejemplo de varios tipos ---
int numeroEntero = 14;
double numeroDecimal = 3.1416;
string cadenaTexto = "Agustín";
bool soyHumano = true;
bool hoyLunes = false;
char inicial = 'A';

// Variables de numeros enteros (int)
int edadProfesor = 26;
int dientes = 32;
int amigos = 5;

// Variables de numeros decimales (double)
double altura = 1.76;
double saldoBanco = 1250.35;
double precioHotDogCostco = 1.50;

// Variables de texto (string)
string diaSemana = "Martes";
string diaActual = "7";
string mesActual = "Julio";
string anioActual = "2026";

// Variables booleanas (bool)
bool lloviendo = false;
bool ganeLoteria = false;
bool tengoMascotas = true;

Console.WriteLine($"¡Soy tu profesor! Mi nombre es {cadenaTexto}, mucho gusto.\n");
Console.WriteLine($"Inicial del nombre: {inicial}\n");

Console.WriteLine("¡IMPRIMIENDO VARIABLES DE NÚMEROS ENTEROS!\n");
Console.WriteLine($"Tu profesor tiene {edadProfesor} años.");
Console.WriteLine($"Tu profesor tiene {dientes} dientes.");
Console.WriteLine($"Tu profesor tiene {amigos} amigos.\n");

Console.WriteLine("¡IMPRIMIENDO VARIABLES DE NÚMEROS DECIMALES!\n");
Console.WriteLine($"Tu profesor mide {altura} metros.");
Console.WriteLine($"Tu profesor tiene {saldoBanco} euros en el banco.");
Console.WriteLine($"Hot Dog en Costco: {precioHotDogCostco} euros.\n");

Console.WriteLine("¡IMPRIMIENDO VARIABLES DE TEXTO!\n");
// Concatenacion clasica con +
Console.WriteLine("Hoy es " + diaSemana + ", " + diaActual + " de " + mesActual + " del " + anioActual + ".");
// Misma idea con interpolacion (mas comoda)
Console.WriteLine($"Hoy es {diaSemana}, {diaActual} de {mesActual} del {anioActual}.");
Console.WriteLine($"El nombre de tu profesor es {cadenaTexto}.\n");

Console.WriteLine("¡IMPRIMIENDO VARIABLES BOOLEANAS!\n");
Console.WriteLine($"¿Hoy esta lloviendo? {lloviendo}");
Console.WriteLine($"¿Me he ganado la loteria? {ganeLoteria}");
Console.WriteLine($"¿Tengo mascotas? {tengoMascotas}");
Console.WriteLine($"¿Soy humano? {soyHumano} | ¿Hoy es lunes? {hoyLunes}");
Console.WriteLine($"Numero de ejemplo: {numeroEntero}, PI aproximado: {numeroDecimal}");

Console.WriteLine("\n¡FIN DEL PROGRAMA!\n");
